packages/dds/tree/src/feature-libraries/chunked-forest/chunkTree.ts

```diff
@@ -226,75 +231,149 @@ export function makePolicy(policy?: Partial<ChunkPolicy>): ChunkPolicy {
 	return withDefaults;
 }
 
-export function shapesFromSchema(
-	schema: StoredSchemaCollection,
-	policy: FullSchemaPolicy,
-): Map<TreeNodeSchemaIdentifier, ShapeInfo> {
-	const shapes: Map<TreeNodeSchemaIdentifier, ShapeInfo> = new Map();
-	for (const identifier of schema.nodeSchema.keys()) {
-		tryShapeFromSchema(schema, policy, identifier, shapes);
-	}
-	return shapes;
+export interface ShapeFromSchemaParameters {
+	/**
+	 * The collection of stored schemas containing node and field definitions.
+	 */
+	readonly schema: StoredSchemaCollection;
+	/**
+	 * Policy from the app for interpreting the stored schema.
+	 */
+	readonly policy: FullSchemaPolicy;
+	/**
+	 * Policy function to determine if a field should be encoded incrementally.
+	 * Incrementally encoding requires the subtree to not start in the middle of a larger uniform chunk.
+	 * Thus returning true from this callback indicates that shapes should not be produced which could
+	 *contain the incremental portion as a part of a larger shape.
+	 */
+	readonly shouldEncodeIncrementally: IncrementalEncodingPolicy;
+	/**
+	 * A cache for shapes which may be read and/or updated. A new (or cleared) cache must be provided
+	 * if recalling this function after any of the parameters except in `ShapeFromSchemaParameters` have changed.
+	 */
+	readonly shapes: Map<TreeNodeSchemaIdentifier, ShapeInfo>;
+}
+
+export interface ShapeFromNodeSchemaParameters extends ShapeFromSchemaParameters {
+	/**
+	 * The identifier of the specific node schema to analyze for shape uniformity.
+	 */
+	readonly nodeSchema: TreeNodeSchemaIdentifier;
+}
+
+export interface ShapeFromFieldSchemaParameters extends ShapeFromSchemaParameters {
+	/**
+	 * The identifier of the specific field schema to analyze for shape uniformity.
+	 */
+	readonly fieldSchema: TreeFieldStoredSchema;
+
+	/**
+	 * The identifier of the parent node schema containing this field.
+	 */
+	readonly parentNodeSchema: TreeNodeSchemaIdentifier;
+	/**
+	 * The field key/name used to identify this field within the parent node.
+	 */
+	readonly key: FieldKey;
```

Comment:
I think it would be better to not extend the interface, and have these parameters be provides as separate parameters from the ShapeFromSchemaParameters.

The ShapeFromSchemaParameters object can be long lived, and reused across many calls, including recursively, if it is not extended with these addition properties. 

If might make sense to still have an interface to group the field description together, like:

```suggestion
/**
 * As TreeFieldStoredSchema with some additional context about where it is in the tree.
 */
export interface FieldSchemaWithContext {
	/**
	 * The specific field schema.
	 */
	readonly fieldSchema: TreeFieldStoredSchema;

	/**
	 * The identifier of the parent node schema containing this field.
	 *
	 * If undefined, this is a root field.
	 */
	readonly parentNodeSchema?: TreeNodeSchemaIdentifier;
	
	/**
	 * The field key/name used to identify this field within the parent node.
	 */
	readonly key: FieldKey;
```